### PR TITLE
Retrofit gabi

### PR DESCRIPTION
**Pull request by gsintrepid**
- Maintained AsyncTask but using retrofit (will create another branch to remove asynctask)

### CODE REVIEW

File: app/src/main/java/io/intrepid/webservicesexercise/GetGitHubUserTask.java [hunk 1]
```diff
@@ -2,41 +2,60 @@
...
+import retrofit2.converter.gson.GsonConverterFactory;
 
 class GetGitHubUserTask extends AsyncTask<String, Void, GitHubUser> {
+    public static final String BASE_URL = "https://api.github.com/";
+    private static final long CONNECTION_TIMEOUT = 15;
+
     private Callback callback;
 
+    private static GithubApi createRestApi() {
+        OkHttpClient.Builder builder = new OkHttpClient.Builder();
```
Comment (weathersboyle) [inline app/src/main/java/io/intrepid/webservicesexercise/GetGitHubUserTask.java]: builder does not need to be declared; can chain everything together

File: app/src/main/java/io/intrepid/webservicesexercise/GetGitHubUserTask.java [hunk 2]
```diff
@@ -2,41 +2,60 @@
...
-            httpURLConnection.disconnect();
+            Call<GitHubUser> call = api.getUser(params[0]);
+            GitHubUser user = call.execute().body();
 
-            Gson gson = new Gson();
-            return gson.fromJson(response.toString(), GitHubUser.class);
+            return user;
         } catch (IOException e) {
-            Timber.e(e);
+            e.printStackTrace();
```
Comment (weathersboyle) [inline app/src/main/java/io/intrepid/webservicesexercise/GetGitHubUserTask.java]: printStackTrace() is useless -- use Timber

**Review by weathersboyle — CHANGES_REQUESTED** [review]
now try it async 👍

**Comment by gsintrepid** [comment]
@weathersboyle Updated! Thanks

**Pull request closed without merging**